op-e2e/actions/interop/interop_test.go

```diff
@@ -543,10 +545,283 @@ func TestInteropFaultProofs(gt *testing.T) {
+
+	// Confirm transaction inclusion
+	rec, err := actors.ChainB.SequencerEngine.EthClient().TransactionReceipt(t.Ctx(), execTx.Hash())
+	require.NoError(t, err)
+	require.NotNil(t, rec)
+
+	// safe head is still behind until we verify cross-safe
+	assertHeads(t, actors.ChainA, 3, 3, 2, 2)
+	assertHeads(t, actors.ChainB, 3, 3, 2, 2)
+	endTimestamp := actors.ChainB.Sequencer.L2Unsafe().Time
+
+	chainAClient := actors.ChainA.Sequencer.RollupClient()
+	chainBClient := actors.ChainB.Sequencer.RollupClient()
+
+	ctx := context.Background()
+	startTimestamp := endTimestamp - 1
+	source, err := NewSuperRootSource(ctx, chainAClient, chainBClient)
+	require.NoError(t, err)
+	start, err := source.CreateSuperRoot(ctx, startTimestamp)
+	require.NoError(t, err)
+	end, err := source.CreateSuperRoot(ctx, endTimestamp)
+	require.NoError(t, err)
+
+	endBlockNumA, err := actors.ChainA.RollupCfg.TargetBlockNumber(endTimestamp)
+	require.NoError(t, err)
+	chain1End, err := chainAClient.OutputAtBlock(ctx, endBlockNumA)
+	require.NoError(t, err)
+
+	endBlockNumB, err := actors.ChainB.RollupCfg.TargetBlockNumber(endTimestamp)
+	require.NoError(t, err)
+	chain2End, err := chainBClient.OutputAtBlock(ctx, endBlockNumB)
+	require.NoError(t, err)
+
+	step1Expected := (&types.TransitionState{
+		SuperRoot: start.Marshal(),
+		PendingProgress: []types.OptimisticBlock{
+			{BlockHash: chain1End.BlockRef.Hash, OutputRoot: chain1End.OutputRoot},
+		},
+		Step: 1,
+	}).Marshal()
+
+	step2Expected := (&types.TransitionState{
+		SuperRoot: start.Marshal(),
+		PendingProgress: []types.OptimisticBlock{
+			{BlockHash: chain1End.BlockRef.Hash, OutputRoot: chain1End.OutputRoot},
+			{BlockHash: chain2End.BlockRef.Hash, OutputRoot: chain2End.OutputRoot},
+		},
+		Step: 2,
+	}).Marshal()
+
+	paddingStep := func(step uint64) []byte {
+		return (&types.TransitionState{
+			SuperRoot: start.Marshal(),
+			PendingProgress: []types.OptimisticBlock{
+				{BlockHash: chain1End.BlockRef.Hash, OutputRoot: chain1End.OutputRoot},
+				{BlockHash: chain2End.BlockRef.Hash, OutputRoot: chain2End.OutputRoot},
+			},
+			Step: step,
+		}).Marshal()
+	}
+
+	// Induce block replacement
+	verifyCrossSafe(t, actors)
+	// assert that the invalid message tx was reorged out
+	_, err = actors.ChainB.SequencerEngine.EthClient().TransactionReceipt(t.Ctx(), execTx.Hash())
+	require.ErrorIs(gt, err, ethereum.NotFound)
+
+	crossSafeSuperRootEnd, err := source.CreateSuperRoot(ctx, endTimestamp)
+	require.NoError(t, err)
+
+	tests := []*transitionTest{
+		{
+			name:               "ClaimNoChange",
+			agreedClaim:        start.Marshal(),
+			disputedClaim:      start.Marshal(),
+			disputedTraceIndex: 0,
+			expectValid:        false,
+			skipChallenger:     true,
+		},
+		{
+			name:               "ClaimDirectToNextTimestamp",
+			agreedClaim:        start.Marshal(),
+			disputedClaim:      end.Marshal(),
+			disputedTraceIndex: 0,
+			expectValid:        false,
+			skipChallenger:     true,
+		},
```

Comment:
Do we need these invalid cases? Not sure they're really hitting anything extra here that isn't covered in the happy path version of them (really they are just checking we actually validate the block hash and don't consider everything valid).

Reply:
No we don't need these. Removing.